pkg/ddc/alluxio/ufs.go

```diff
@@ -90,6 +93,22 @@ func (e *AlluxioEngine) ShouldUpdateUFS() (ufsToUpdate *utils.UFSToUpdate) {
 	ufsToUpdate = utils.NewUFSToUpdate(dataset)
 	ufsToUpdate.AnalyzePathsDelta()
 
+	// 3. for hostpath ufs mount, check if all mountpoints have been mounted
+	const envVar = "FLUID_ENABLE_REMOUNT_DURING_SYNC"
+	if os.Getenv(envVar) != ""{
+		unmountedPaths, err := e.FindUnmountedUFS()
+		if err != nil {
+			e.Log.Error(err, "Failed in finding unmounted ufs")
+			return
+		}
+		if len(unmountedPaths) != 0{
+			ufsToUpdate.AddMountPaths(unmountedPaths)
```

Comment:
if we don't check AddedMountPaths before adding it, is it potential risk to cause the slice of `u.toAdd` leak?

Reply:
Add as suggested.